build.sh

```diff
@@ -28,11 +28,14 @@ export DOTNET_INSTALL_DIR="$REPOROOT/.dotnet"
 echo DOTNET_INSTALL_DIR=$DOTNET_INSTALL_DIR
 export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
 
-mkdir $DOTNET_INSTALL_DIR
-curl -sSL https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain/dotnet-install.sh | bash /dev/stdin --install-dir $DOTNET_INSTALL_DIR
+if [ ! -d "$DOTNET_INSTALL_DIR" ]; then 
+	mkdir $DOTNET_INSTALL_DIR
+	curl -sSL https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain/dotnet-install.sh | bash /dev/stdin --install-dir $DOTNET_INSTALL_DIR
```

Comment:
This should not be inside of the if block. It will create non-deterministic builds here by locking the CLI version to whatever is in the directory. a clean clone will get a different CLI version. If you want a stable version, specify it to the script. Otherwise, let the script check for latest on each build.